ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,836 @@
+				ForceNew:         true,
+				Default:          "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT",
+				DiffSuppressFunc: applyOnce,
+			},
+
+			"redundant_network": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			"unbonded_network": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			"public_bandwidth": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          20000,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			"memory": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			// // Base for Gateway vlans array
+			// "storage_groups": {
+			// 	Type:     schema.TypeList,
+			// 	Optional: true,
+			// 	ForceNew: true,
+			// 	Elem: &schema.Resource{
+			// 		Schema: map[string]*schema.Schema{
+			// 			"array_type_id": {
+			// 				Type:     schema.TypeInt,
+			// 				Required: true,
+			// 			},
+			// 			"hard_drives": {
+			// 				Type:     schema.TypeList,
+			// 				Elem:     &schema.Schema{Type: schema.TypeInt},
+			// 				Required: true,
+			// 			},
+			// 			"array_size": {
+			// 				Type:     schema.TypeInt,
+			// 				Optional: true,
+			// 			},
+			// 			"partition_template_id": {
+			// 				Type:     schema.TypeInt,
+			// 				Optional: true,
+			// 			},
+			// 		},
+			// 	},
+			// 	DiffSuppressFunc: applyOnce,
+			// },
+
+			// Quote based provisioning only
+			"quote_id": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Quote based provisioning, Monthly
+			"public_vlan_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			// Quote based provisioning, Monthly
+			"private_vlan_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"public_ipv4_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"private_ipv4_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipv6_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+
+			"ipv6_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"ipv6_address_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			// SoftLayer does not support public_ipv6_subnet configuration in vm creation. So, public_ipv6_subnet
+			// is defined as a computed parameter.
+			"public_ipv6_subnet": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vlan_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"private_network_only": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			"associated_vlans": {
```

Comment:
Can we read back this info and set in the read method? This can be useful while importing an already existing resource in SoftLayer and bringing it under terraform management. Since this is a unique feature and widely usable can we also add a test for this? Is this field not updatable? I don't see this being used in the update method. If not we should set to ForceNew on this?

Reply:
@ashishth09 working in those observations, also I will be adding the code to run the config portion of the resource (Vyatta configuration later today), thank you